server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3066,7 +3071,10 @@ private void deleteSnapshotsFromRepository(
                     ActionListener.wrap(updatedRepoData -> {
                         logger.info("snapshots {} deleted", snapshotIds);
                         removeSnapshotDeletionFromClusterState(deleteEntry, null, updatedRepoData);
-                    }, ex -> removeSnapshotDeletionFromClusterState(deleteEntry, ex, repositoryData))
+                    }, ex -> {
+                        logger.debug(() -> "failed to complete snapshot deletion [" + deleteEntry + "]", ex);
+                        removeSnapshotDeletionFromClusterState(deleteEntry, ex, repositoryData);
```

Comment:
@DaveCTurner 7.17 does not have the `SnapshotDeleteListener`. I believe this is where the logging line should go. But would appreciate your confirmation. Thanks!